flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -543,6 +542,144 @@ public void disableAutoTypeRegistration() {
+		result = prime
+				* result
+				+ ((registeredTypesWithKryoSerializerClasses == null) ? 0
+						: registeredTypesWithKryoSerializerClasses.hashCode());
+		result = prime
+				* result
+				+ ((registeredTypesWithKryoSerializers == null) ? 0
+						: registeredTypesWithKryoSerializers.hashCode());
+		result = prime * result + (useClosureCleaner ? 1231 : 1237);
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (obj == null) {
+			return false;
+		}
+		if (getClass() != obj.getClass()) {
+			return false;
+		}
+		ExecutionConfig other = (ExecutionConfig) obj;
+		if (defaultKryoSerializerClasses == null) {
+			if (other.defaultKryoSerializerClasses != null) {
+				return false;
+			}
+		} else if (!defaultKryoSerializerClasses
+				.equals(other.defaultKryoSerializerClasses)) {
+			return false;
+		}
+		if (defaultKryoSerializers == null) {
+			if (other.defaultKryoSerializers != null) {
+				return false;
+			}
+		} else if (!defaultKryoSerializers.equals(other.defaultKryoSerializers)) {
+			return false;
+		}
+		if (disableAutoTypeRegistration != other.disableAutoTypeRegistration) {
+			return false;
+		}
+		if (executionMode != other.executionMode) {
+			return false;
+		}
+		if (forceAvro != other.forceAvro) {
+			return false;
+		}
+		if (forceKryo != other.forceKryo) {
+			return false;
+		}
+		if (globalJobParameters == null) {
+			if (other.globalJobParameters != null) {
+				return false;
+			}
+		} else if (!globalJobParameters.equals(other.globalJobParameters)) {
+			return false;
+		}
+		if (numberOfExecutionRetries != other.numberOfExecutionRetries) {
+			return false;
+		}
+		if (objectReuse != other.objectReuse) {
+			return false;
+		}
+		if (parallelism != other.parallelism) {
+			return false;
+		}
+		if (printProgressDuringExecution != other.printProgressDuringExecution) {
+			return false;
+		}
+		if (registeredKryoTypes == null) {
+			if (other.registeredKryoTypes != null) {
+				return false;
+			}
+		} else if (!registeredKryoTypes.equals(other.registeredKryoTypes)) {
+			return false;
+		}
+		if (registeredPojoTypes == null) {
+			if (other.registeredPojoTypes != null) {
+				return false;
+			}
+		} else if (!registeredPojoTypes.equals(other.registeredPojoTypes)) {
+			return false;
+		}
+		if (registeredTypesWithKryoSerializerClasses == null) {
+			if (other.registeredTypesWithKryoSerializerClasses != null) {
+				return false;
+			}
+		} else if (!registeredTypesWithKryoSerializerClasses
+				.equals(other.registeredTypesWithKryoSerializerClasses)) {
+			return false;
+		}
+		if (registeredTypesWithKryoSerializers == null) {
+			if (other.registeredTypesWithKryoSerializers != null) {
+				return false;
+			}
+		} else if (!registeredTypesWithKryoSerializers
+				.equals(other.registeredTypesWithKryoSerializers)) {
+			return false;
+		}
+		if (useClosureCleaner != other.useClosureCleaner) {
+			return false;
+		}
+		return true;
+	}
```

Comment:
Thats more than 100 LOC for a test case.
Maybe it would be better to do these equality checks in the test case?
Also, I fear that people won't maintain the two methods when they add new fields to the `ExecutionConfig`